### DISCUSSION TITLE
Is a local login method wanted?

### DISCUSSION THREAD

**Discussion by Cilenco** [outline/outline#3180] [Ideas]
Sorry to open another one of these questions but since it is pretty thought through I thought I try my luck!
Since I'm not sure if this is a feature which is wanted by the core team I wanted to ask about this first.

I would be happy to work on a local login. Ideas would include:

- New login in form for E-Mail and password
- Adjustment of the user profile scene to change the current password (for all users)
- Existing users can create a password and new useres are able to link a supported account
- Creating a new scene for "I forgot my password" which sends an E-Mail to a provided address
- Users invited by E-Mail are asked to create a password the first time they use the magic link
- New env variables will be introduced to enable/disable local login and a default admin user + password
- New useres can be invited by admins through the existing E-Mail feature
- Local users will be added to a virtual team (maybe just named local)

Of course I'm open to any discussions!

**Comment by ash-development** [comment]
I can say that I am 100% interested in this. Personally for my uses, using Slack or G Suite/Google Workspace/Whatever it's called nowadays simply aren't a viable option.

**Reply by okase** [reply]
I agree. I feel like having a simple email auth will even make this app more enticing for people to try since they won't have to struggle setting up the other integrations.

**Comment by rusty1281** [comment]
+1 on the feature. Last year in June I have tested this platform. When I saw the options for login I said to myself this is not for me. On top of this, files were pointed to minio or S3, again not ideal. I see that OIDC have been offered now, and file attachments are part of the Outline instance. Move in the right direction. Still not sure why a simple sign-in is missing.

Think I will wait for it before revisiting this.

**Comment by tonycoculuzzi** [comment]
+1 as well. A local login authentication provider should be the default, with Slack, GSuite and Microsoft authentication being optional. The whole installation process has been complicated by simply requiring at least one of these three external services for login.

**Comment by navjotjsingh** [comment]
Totally Agree. Setting up external authentication providers can be a painful process and not always viable. Maybe further down the road, the option to add Two-factor authentication will help secure the simple login.

**Comment by tillsanders** [comment]
This is a big missing feature in my opinion. Not being able to login with classic email/password is a real problem, especially in regard to data privacy. I would love to see this feature!

**Comment by ggtylerr** [comment]
I recently found out about this project, and although it looks nice, the *requirement* to use integrations is absolutely annoying. Biggest thing is (for me at least) I don't use Google Workspace, Slack, or Azure, and now I'm required to just for this. This is a standard across literally every service and there's a reason for that.

**Reply by tillsanders** [reply]
I agree, password-based authentication is a must for enterprise usage.

**Reply by ash-development** [reply]
This!!

**Comment by tidusvn05** [comment]
my company is not use google workspace, azure and slack, so we very difficult use outline.
about this toptic, any solution from fork project ?

**Reply by almereyda** [reply]
You can always use your own OpenID Connect Identity Provider, such as Keycloak, Authentik and similar offers.

**Comment by nsauter** [comment]
+1 For me. Did anybody of the devs say someting about a plan to implement this?

**Comment by mweth** [comment]
+1  This lack makes dev and home deployments so much harder than it has to be.

**Comment by ovizii** [comment]
Any news on this or was this just a teaser? ;-)

**Comment by vxe4** [comment]
Yup. 100% on board for this feature, as I host a personal directory (well, once they address my discussion) and find no need for an integration. Approved.
